internal/metarepos/storage.go

```diff
@@ -585,6 +593,49 @@ func (ms *MetadataStorage) updateLogStream(ls *varlogpb.LogStreamDescriptor) err
 	return err
 }
 
+func (ms *MetadataStorage) willBeSealed(ls *varlogpb.LogStreamDescriptor) bool {
+	pre, cur := ms.getStateMachine()
+
+	reports, ok := cur.LogStream.UncommitReports[ls.LogStreamID]
+	if !ok {
+		tmp, ok := pre.LogStream.UncommitReports[ls.LogStreamID]
+		if !ok {
+			return false
+		}
+
+		reports = proto.Clone(tmp).(*mrpb.LogStreamUncommitReports)
+	}
+
+	lastCommitted := ms.getLastCommittedLLSN(ls.TopicID, ls.LogStreamID)
+	for _, r := range ls.Replicas {
+		o := reports.Replicas[r.StorageNodeID]
+		if lastCommitted < o.UncommittedLLSNOffset {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (ms *MetadataStorage) getLastCommittedLLSN(topicID types.TopicID, lsID types.LogStreamID) types.LLSN {
+	crs := ms.GetLastCommitResults()
+	if crs == nil {
+		return types.InvalidLLSN
+	}
```

Comment:
It means no commit history in the cluster. This may be when the cluster is initialized.
L629 means that ls has never been committed yet, If ls exists in LogStreamDesc.

Reply:
In the chain of calling `updateLogStream`, `willBeSealed`, and `getLastCommittedLLSN`, I think that a nil result from GetLastCommitResults usually seems to be unreachable. Do I understand right?

Reply:
If a cluster is created and there is no committed log, nil is returned.
In `TestMRUpdateLogStream`, the result from GetLastCommitResults is nil because it seals and updates the empty logstream.

Reply:
Thank you for the explanation.